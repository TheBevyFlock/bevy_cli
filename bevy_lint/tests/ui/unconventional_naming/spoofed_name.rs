//! A test that ensures a plugin whose name is "spoofed" with `use T as F` does not sneak past the
//! lint.

#![feature(register_tool)]
#![register_tool(bevy)]
#![deny(bevy::unconventional_naming)]
//~^ NOTE: the lint level is defined here

use bevy::prelude::*;

mod bar {
    pub mod baz {
        pub struct Foo;
        //~^ ERROR: unconventional type name for a `Plugin`
        //~| NOTE: structures that implement `Plugin` should end in `Plugin`
        //~| HELP: rename `Foo`
    }
}

// We try to be sneaky, but it doesn't work.
use self::bar::baz::Foo as FooPlugin;

//~v NOTE: `Plugin` implemented here
impl Plugin for FooPlugin {
    fn build(&self, _app: &mut App) {}
}

fn main() {
    App::new().add_plugins(FooPlugin);
}
