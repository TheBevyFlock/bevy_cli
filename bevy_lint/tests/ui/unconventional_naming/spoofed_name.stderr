error: unconventional type name for a `Plugin`
  --> tests/ui/unconventional_naming/spoofed_name.rs:13:20
   |
13 |         pub struct Foo;
   |                    ^^^ help: rename `Foo`: `FooPlugin`
   |
note: structures that implement `Plugin` should end in `Plugin`
  --> tests/ui/unconventional_naming/spoofed_name.rs:13:20
   |
13 |         pub struct Foo;
   |                    ^^^
note: `Plugin` implemented here
  --> tests/ui/unconventional_naming/spoofed_name.rs:24:1
   |
24 | / impl Plugin for FooPlugin {
25 | |     fn build(&self, _app: &mut App) {}
26 | | }
   | |_^
note: the lint level is defined here
  --> tests/ui/unconventional_naming/spoofed_name.rs:6:9
   |
6  | #![deny(bevy::unconventional_naming)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 1 previous error

